dev/WindowsAppSDK_Insights/WindowsAppSDKInsights.h

```diff
@@ -10,8 +10,49 @@
 #error "WIL Tracelogging.h must not be explicitly included when including this file"
 #endif
 
-char* WINDOWSAPPSDK_PACKAGE_VER();
-char* WINDOWSAPPSDK_EXPERIMENTATION_LEVEL();
+#include <wil/resource.h>
+#include <string>
+
+#define _WINDOWSAPPSDK_PACKAGE_VER_RESOURCEID             10000
+#define _WINDOWSAPPSDK_EXPERIMENTATION_FLAG_RESOURCEID    10001
```

Comment:
These are part of your API? I see nothing about INSIGHTS in their name., despite them being very much part of the Insights DLL and resources. Further, these are global scope symbols. At a minimum they need to be suitably scoped.

```
#define _WINDOWSAPPSDK_INSIGHTS_RESOURCEID_PACKAGE_VERSION   10000
#define _WINDOWSAPPSDK_INSIGHTS_RESOURCEID_CHANNEL           10001
```

But I see a broader range of issues, especially as this is an API and not just implementation detail.
1. Very C - global symbols, #defines, more
2. Unscoped names - almost no symbol indicates it's part of Insights
3. load_resource_string swallows errors...but (sometimes) throws exceptions - I don't see any HRESULT or throw, and yet std::string is used so it could throw exceptions. Is your intent to never throw and never return an error code, just return values or 'dead-values' (null, 0, etc) if something doesn't go perfectly right?
4. Assumption max-path-length-for-resource-dll is <256 chars - that's rather dated by today's standards. We need to be prepared to handle longer paths

Even if this is internal (not a public API) we can (and should) write Modern C++. No need to be restricted to Win32 C export rules in a header file meant for (very) C++ callers :-)

Here's one potential idea:

```
namespace Microsoft.Windows.Something.Insights
{
class RuntimeInformation
{
public:
    static std::string GetVersion()
    {
        const uint32_t c_versionId{ 10000 };
        static std::string value{ LoadStringFromResource(c_versionId) };
        return value;
    }

    static std::string GetChannel()
    {
        const uint32_t c_channelId{ 10001 };
        static std::string value{ LoadStringFromResource(c_channelId) };
        return value;
    }

private:
    static std::string LoadStringFromResource(uint32_t id)
    {
        static wil::unique_hmodule module = std::move(LoadResourceModule());

        // We retrieve known controlled data so we know it's max length and expect it to exist
        const int c_valueMaxLength{ 100 };
        char value[c_valueMaxLength]{};
        THROW_IF_WIN32_BOOL_FALSE_MSG(::LoadStringA(module.get(), id, value, ARRAYSIZE(value)), "Failed to load resource string id:%d", id);
        return std::string(value);
    }

    static std::string LoadResourceModule()
    {
        PCWSTR filename{ "Microsoft.WindowsAppSDK.Insights.Resource.dll" };
        wil::unique_hmodule module(::LoadLibraryA(filename));
        THROW_HR_IF_NULL_MSG(HRESULT_FROM_WIN32(GetLastError()), module, "Failed to load %hs", filename);
        return module;
    }
};
}
```

where usage is `auto version{ Microsoft::Windows::Something::Insights::RuntimeInformation::GetVersion() };` and `GetChannel`.

Symbols are scoped. Everything's in an `Insights` namespace. The class trick with static methods gives scoping so only `GetVersion` and `GetChannel` are exposed, all the implementation is private. Implementation details like what ids are used for what resources (or even that resources are used at all) is hidden inside the implementation, preventing leakage into every callsite.

This assumes an exception-based error handling strategy. If you want an error code base strategy you'd need functions returning HRESULT and moving retvals to out parameters. If you want a no-fail API then you need to replace std::string, e.g. return PCSTR to a static const value (instead of std::string), use fixed-length static buffers or static pointer to dynamically allocated memory (e.g. `std::unique_ptr<WCHAR[]>`), and so forth.

This is one of multiple possible solutions. The important point is to avoid unscoped global symbols, and to use a consistent error strategy.